sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CustomShuffleReaderExec.scala

```diff
@@ -124,6 +104,24 @@ case class CustomShuffleReaderExec private(
     Map("numSkewedPartitions" -> metrics)
   }
 
+  private def sendPartitionDataSizeMetrics(executionId: String): Unit = {
+    val mapStats = shuffleStage.get.mapStats.bytesByPartitionId
+    partitionSpecs.foreach {
+      case CoalescedPartitionSpec(startReducerIndex, endReducerIndex) =>
+        val dataSize = startReducerIndex.until(endReducerIndex).map(mapStats(_)).sum
+        metrics("partitionDataSize").set(dataSize)
```

Comment:
we should call `add`, not `set`. The SQL metrics itself should contain the sum value.